Traceback (most recent call last):
  File "/Users/mikemorris/opt/anaconda3/envs/UTCDW-env/lib/python3.10/site-packages/jupyter_cache/executors/utils.py", line 58, in single_nb_execution
    executenb(
  File "/Users/mikemorris/opt/anaconda3/envs/UTCDW-env/lib/python3.10/site-packages/nbclient/client.py", line 1305, in execute
    return NotebookClient(nb=nb, resources=resources, km=km, **kwargs).execute()
  File "/Users/mikemorris/opt/anaconda3/envs/UTCDW-env/lib/python3.10/site-packages/jupyter_core/utils/__init__.py", line 168, in wrapped
    return loop.run_until_complete(inner)
  File "/Users/mikemorris/opt/anaconda3/envs/UTCDW-env/lib/python3.10/asyncio/base_events.py", line 649, in run_until_complete
    return future.result()
  File "/Users/mikemorris/opt/anaconda3/envs/UTCDW-env/lib/python3.10/site-packages/nbclient/client.py", line 705, in async_execute
    await self.async_execute_cell(
  File "/Users/mikemorris/opt/anaconda3/envs/UTCDW-env/lib/python3.10/site-packages/nbclient/client.py", line 1058, in async_execute_cell
    await self._check_raise_for_error(cell, cell_index, exec_reply)
  File "/Users/mikemorris/opt/anaconda3/envs/UTCDW-env/lib/python3.10/site-packages/nbclient/client.py", line 914, in _check_raise_for_error
    raise CellExecutionError.from_cell_and_msg(cell, exec_reply_content)
nbclient.exceptions.CellExecutionError: An error occurred while executing the following cell:
------------------
# do the significance testing
reject_null_raw = test_significance(ds_tn_hist_raw, ds_tn_future_raw)
reject_null_sds = test_significance(ds_tn_hist_sds, ds_tn_future_sds)
------------------


[0;31m---------------------------------------------------------------------------[0m
[0;31mAttributeError[0m                            Traceback (most recent call last)
Cell [0;32mIn[14], line 2[0m
[1;32m      1[0m [38;5;66;03m# do the significance testing[39;00m
[0;32m----> 2[0m reject_null_raw [38;5;241m=[39m [43mtest_significance[49m[43m([49m[43mds_tn_hist_raw[49m[43m,[49m[43m [49m[43mds_tn_future_raw[49m[43m)[49m
[1;32m      3[0m reject_null_sds [38;5;241m=[39m test_significance(ds_tn_hist_sds, ds_tn_future_sds)

Cell [0;32mIn[13], line 9[0m, in [0;36mtest_significance[0;34m(ds_hist, ds_future, alpha)[0m
[1;32m      4[0m tstats, pvals [38;5;241m=[39m stats[38;5;241m.[39mttest_ind_from_stats(ds_hist[[38;5;124m'[39m[38;5;124mmean_tn[39m[38;5;124m'[39m], ds_hist[[38;5;124m'[39m[38;5;124mstd_tn[39m[38;5;124m'[39m], ds_hist[[38;5;124m'[39m[38;5;124mneff[39m[38;5;124m'[39m],
[1;32m      5[0m                                            ds_future[[38;5;124m'[39m[38;5;124mmean_tn[39m[38;5;124m'[39m], ds_future[[38;5;124m'[39m[38;5;124mstd_tn[39m[38;5;124m'[39m], ds_future[[38;5;124m'[39m[38;5;124mneff[39m[38;5;124m'[39m],
[1;32m      6[0m                                            equal_var [38;5;241m=[39m [38;5;28;01mFalse[39;00m)
[1;32m      8[0m [38;5;66;03m# get new threshold using alpha_FDR and get array of 1's where p < alpha_FDR (and 0's otherwise)[39;00m
[0;32m----> 9[0m alpha_FDR [38;5;241m=[39m [43mfdr[49m[43m([49m[43mpvals[49m[43m,[49m[43m [49m[43malpha[49m[43m [49m[38;5;241;43m=[39;49m[43m [49m[43malpha[49m[43m)[49m
[1;32m     10[0m reject_null [38;5;241m=[39m np[38;5;241m.[39mwhere(pvals [38;5;241m<[39m alpha_FDR, [38;5;241m1[39m, [38;5;241m0[39m)
[1;32m     12[0m [38;5;28;01mreturn[39;00m reject_null

Cell [0;32mIn[12], line 9[0m, in [0;36mfdr[0;34m(p, alpha)[0m
[1;32m      7[0m dims [38;5;241m=[39m p[38;5;241m.[39mshape
[1;32m      8[0m [38;5;28;01mif[39;00m [38;5;28mlen[39m(dims) [38;5;241m>[39m [38;5;241m1[39m:
[0;32m----> 9[0m     p [38;5;241m=[39m [43mp[49m[38;5;241;43m.[39;49m[43mflatten[49m()
[1;32m     10[0m sortp [38;5;241m=[39m np[38;5;241m.[39msort(p)
[1;32m     11[0m N [38;5;241m=[39m [38;5;28mlen[39m(p)

File [0;32m~/opt/anaconda3/envs/UTCDW-env/lib/python3.10/site-packages/xarray/core/common.py:278[0m, in [0;36mAttrAccessMixin.__getattr__[0;34m(self, name)[0m
[1;32m    276[0m         [38;5;28;01mwith[39;00m suppress([38;5;167;01mKeyError[39;00m):
[1;32m    277[0m             [38;5;28;01mreturn[39;00m source[name]
[0;32m--> 278[0m [38;5;28;01mraise[39;00m [38;5;167;01mAttributeError[39;00m(
[1;32m    279[0m     [38;5;124mf[39m[38;5;124m"[39m[38;5;132;01m{[39;00m[38;5;28mtype[39m([38;5;28mself[39m)[38;5;241m.[39m[38;5;18m__name__[39m[38;5;132;01m!r}[39;00m[38;5;124m object has no attribute [39m[38;5;132;01m{[39;00mname[38;5;132;01m!r}[39;00m[38;5;124m"[39m
[1;32m    280[0m )

[0;31mAttributeError[0m: 'DataArray' object has no attribute 'flatten'

